File: backend/app/Models/Category.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Category extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'name',
        'main_category_id',
    ];

    public function projects()
    {
        return $this->hasMany(Project::class);
    }

    public function evaluators()
    {
        return $this->belongsToMany(Evaluator::class, 'evaluator_categories');
    }

    public static function mainCategories()
    {
        return self::whereNull('main_category_id')->get();
    }

    public function mainCategory()
    {
        return $this->belongsTo(Category::class, 'main_category_id');
    }

    public function subCategories()
    {
        return $this->hasMany(Category::class, 'main_category_id');
    }
}

File: backend/app/models/category.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.database import Base

class Category(Base):
    __tablename__ = "categories"
    
    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(255), nullable=False)
    main_category_id = Column(Integer, ForeignKey("categories.id"), nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
    deleted_at = Column(DateTime(timezone=True), nullable=True)
    
    projects = relationship("Project", back_populates="category")
    evaluators = relationship("Evaluator", secondary="evaluator_categories", back_populates="categories")
    main_category = relationship("Category", remote_side=[id], backref="sub_categories")
